### PR TITLE
Fix bug in validating unused bits

### PR DESCRIPTION
**Pull request by buyaa-n**
Fix bug in validating unused bits found with [Base64Url fuzzer](https://github.com/dotnet/runtime/pull/106509) 

The input that failed the fuzzer is `Base64Url.IsValid("SM=")` which supposed to return false, invalid input

### CODE REVIEW

**Comment by MihaZupan** [comment]
Looks like there are still some unhandled cases: https://github.com/MihuBot/runtime-utils/issues/562
```
// Base64UrlFuzzer
System.Exception: Expected=Done Actual=InvalidData
   at DotnetFuzzing.Assert.<Equal>g__Throw|0_0[T](T expected, T actual) in D:\a\runtime-utils\runtime-utils\Runner\runtime\src\libraries\Fuzzing\DotnetFuzzing\Assert.cs:line 18
   at DotnetFuzzing.Assert.Equal[T](T expected, T actual) in D:\a\runtime-utils\runtime-utils\Runner\runtime\src\libraries\Fuzzing\DotnetFuzzing\Assert.cs:line 14
   at DotnetFuzzing.Fuzzers.Base64UrlFuzzer.FuzzTarget(ReadOnlySpan`1 bytes) in D:\a\runtime-utils\runtime-utils\Runner\runtime\src\libraries\Fuzzing\DotnetFuzzing\Fuzzers\Base64UrlFuzzer.cs:line 109
   at SharpFuzz.Fuzzer.LibFuzzer.Run(ReadOnlySpanAction action, Boolean ignoreExceptions)
==8932== ERROR: libFuzzer: deadly signal
NOTE: libFuzzer has rudimentary signal handlers.
      Combine libFuzzer with AddressSanitizer or similar for better crash reports.
SUMMARY: libFuzzer: deadly signal
MS: 1 CrossOver-; base unit: 122a96fbeb0d3493dcb52f34d0b7bfecbec11036
0x73,0xa,0x45,0x73,0xa,0xa,0x3d,0x3d,
s\012Es\012\012==
artifact_prefix='./'; Test unit written to Base64UrlFuzzer-artifact-2
Base64: cwpFcwoKPT0=
```

**Comment by MihaZupan** [comment]
@MihuBot fuzz Base64Url -combineWith #106509

**Comment by buyaa-n** [comment]
> @MihuBot fuzz Base64Url -combineWith #106509

Its really cool that we can run a fuzzer with a given PR, thanks!

File: src/libraries/System.Private.CoreLib/src/System/Buffers/Text/Base64Url/Base64UrlValidator.cs [hunk 1]
```diff
@@ -94,33 +94,33 @@ public bool ValidateAndDecodeLength(char lastChar, int length, int paddingCount,
...
+
+                if (remainder > 0)
                 {
-                    // unused lower bits are not 0, reject input
-                    decodedLength = 0;
-                    return false;
+                    int decoded = default(Base64UrlDecoderByte).DecodingMap[lastChar];
+                    switch (remainder)
+                    {
+                        case 1: return false; // 1 byte is not decode able => invalid.
```
Comment (stephentoub) [inline src/libraries/System.Private.CoreLib/src/System/Buffers/Text/Base64Url/Base64UrlValidator.cs:120]:
```suggestion
                        case 1: return false; // 1 byte is not decodable => invalid.
```

**Review by stephentoub — APPROVED** [review]

**Pull request merged**